deploy.py

```diff
@@ -272,776 +286,471 @@ def notarizefile(filetonotarize, username, password):
-                    if not os.path.exists('tools/linuxdeployqt'):
-                        os.mkdir('tools/linuxdeployqt')
-
-                    resultCode, resultOutput = execute('cd tools/linuxdeployqt; '
-                                                       'curl -LJO '
-                                                       'https://github.com/probonopd/linuxdeployqt/releases/download/6/'
-                                                       'linuxdeployqt-6-x86_64.AppImage')
-
-                    if resultCode:
-                        endmessage(False, f'unable to download linuxdeployqt.\r\n\r\n{resultOutput}\r\n')
-                        exit(1)
-
-                    resultCode, resultOutput = execute('chmod +x tools/linuxdeployqt/linuxdeployqt-6-x86_64.AppImage')
-
-                    if resultCode:
-                        endmessage(False, f'unable to set permissions on linuxdeployqt.\r\n\r\n{resultOutput}\r\n')
-                        exit(1)
-
-                    endmessage(True)
-
-                    linuxdeployqt = 'tools/linuxdeployqt/linuxdeployqt-6-x86_64.AppImage'
-
-                    appimagetool = args.appimagetool
-
-                    if not os.path.isfile(appimagetool):
-                        startmessage('Downloading appimagetool...')
-
-                        if not os.path.exists('tools/appimagetool'):
-                            os.mkdir('tools/appimagetool')
-
-                        resultCode, resultOutput = execute('cd tools/appimagetool; '
-                                                           'curl -LJO '
-                                                           'https://github.com/AppImage/AppImageKit/releases/download/continuous/'
-                                                           'appimagetool-x86_64.AppImage')
-
-                        if resultCode:
-                            endmessage(False, f'unable to download appimagetool.\r\n\r\n{resultOutput}\r\n')
-                            exit(1)
-
-                        resultCode, resultOutput = execute('chmod +x tools/appimagetool/appimagetool-x86_64.AppImage')
-
-                        if resultCode:
-                            endmessage(False, f'unable to set permissions on appimagetool.\r\n\r\n{resultOutput}\r\n')
-                            exit(1)
-
-                        appimagetool = 'tools/appimagetool/appimagetool-x86_64.AppImage'
-
-                        endmessage(True)
-
-                        # remove previous deployment files and copy current binaries
-
-                        startmessage('Setting up deployment directory...')
-
-                        if os.path.exists(f'bin/{buildArch}/Deploy/AppImage/'):
-                            shutil.rmtree(f'bin/{buildArch}/Deploy/AppImage/')
-
-                        if os.path.exists(f'deployment'):
-                            shutil.rmtree(f'deployment')
+                    # FIXME: We require linux_deploy_qt as argument, and then download it?
```

Comment:
This is definitely out of order and needs to be fixed.

Reply:
Yeah, I changed some stuff in a recent commit and it looks like I completely screwed that up. Oops.

Reply:
Should this be another ticket, since this PR closed?

Reply:
@AaronDMarasco Yeah, open another ticket for this, I'd rather have lots of tickets with one issue per ticket than a "super ticket", makes it easier for us to track what's going on, what needs fixing and so on.  So if we break something, we can trace it down to a commit and not go fishing through all the changes that were made on a "super" commit.  Does that make sense to you?

And FYI, The deploy script is working perfectly with your changes on all platforms.

Reply:
I've fixed this and pushed the fix.  Now you can supply appimage/linuxdeployqt and it will attempt to use that, otherwise it will now download correctly, fixing the broken logic.